docs/concepts/macros/sqlmesh_macros.md

```diff
@@ -1210,26 +1210,25 @@ SELECT
 FROM table
 ```
 
-#### Vararg and kwarg arguments
-Varargs (variable arguments) and kwargs (keyword arguments) are possible to use in Python macros definitions. Kwargs can be used in SQL with the property assignment operator `:=`. 
+#### Macro function arguments
+
+Most macro functions provide arguments so users can supply custom values when the function is called.
+
+Arguments can be provided by position if none are skipped. For example, consider the `add_args()` function - it has three arguments with default values provided in the function definition:
 
 ```python linenums="1"
 from sqlmesh import macro
 
 @macro()
-def kwarg_macro(evaluator, a, b=None, c=None):
-  if b:
-    return b
-  if c:
-    return c
-  return a
+def add_args(evaluator, argument_1=1, argument_2=2, argument_3=3):
```

Comment:
this definition is not right, your defaults are int, but they won't be integers unless you add a type annotation `:int`. as is, arguments will be sqlglot expressions

Reply:
Thanks, I'll update and add more information about typing. Is our general position that macro arg types are required?

Question: why is this rendering correctly without the types?

Macro definition:
```
@macro()
def add_args(evaluator, argument_1=1, argument_2=2, argument_3=3):
    return argument_1 + argument_2 + argument_3
```
Call in model:
```
SELECT
    item_id,
    COUNT(DISTINCT id) AS num_orders,
    @add_args(argument_2 := 1) as add_args,
FROM
    sqlmesh_example.incremental_model
GROUP BY item_id
```
Rendered query:
```
❯ sqlmesh render sqlmesh_example.full_model
SELECT                                                                                                                                                                                                                                                        
  "incremental_model"."item_id" AS "item_id",                                                                                                                                                                                                                 
  COUNT(DISTINCT "incremental_model"."id") AS "num_orders",                                                                                                                                                                                                   
  5 AS "add_args"                                                                                                                                                                                                                                             
FROM "db"."sqlmesh__sqlmesh_example"."sqlmesh_example__incremental_model__1732922178" AS "incremental_model" /* db.sqlmesh_example.incremental_model */                                                                                                       
GROUP BY                                                                                                                                                                                                                                                      
  "incremental_model"."item_id" 
```

Reply:
Answer: i got "luck" - in this specific example, the behavior of the sqlglot expressions happens to result in the same output as if types had been present